pkg/transfers/router_test.go

```diff
@@ -314,8 +314,32 @@ func TestRouter__MissingDestination(t *testing.T) {
 }
 
 func TestRouter__validateAmount(t *testing.T) {
-	if err := validateAmount(client.Amount{}); err == nil {
+	amt := client.Amount{
+		Currency: "USD",
+		Value:    112,
+	}
+	if err := validateAmount(amt); err != nil {
+		t.Errorf("expected no error: %v", err)
+	}
+
+	// invalid currency
+	amt.Currency = "ZZZ"
+	if err := validateAmount(amt); err == nil {
```

Comment:
Nit:

Switching to `t.Fatal` and unnesting the if blocks
```go
	err := validateAmount(amt)
	if err == nil {
		t.Fatal("expected error")
	}
	if !strings.Contains(err.Error(), "unexpected currency") {
		t.Fatal("unexpected error: %v", err)
	}
```
or if you want to stick with `t.Error`, use an `else if`
```go
	err := validateAmount(amt)
	if err == nil {
		t.Error("expected error")
	} else if !strings.Contains(err.Error(), "unexpected currency") {
		t.Errorf("unexpected error: %v", err)
	}
```

Reply:
Updated to the first pattern.